### DISCUSSION TITLE
Use gem5 stats file to find data hazard in code

### DISCUSSION THREAD

**Discussion by sagorchandro** [gem5/gem5#1889] [Q&A] [answered]
I am new and using gem5 for last few days. I am trying to execute the following code and it works fine. It also generate the stats file for the most recent execution. In this code a data hazard is present as result1 variable is dependent on operation 1 while it is being used in operation 2.

#include <stdio.h>
int main() {
int x, y, z; 
int result1, result2, result3;
printf("Enter the values:");
scanf("%d%d%d",&x,&y,&z);
result1 = x + y; 
result2 = result1-z;
result3 = result1 & z;
printf("Result1: %d\n",result1);
printf("Result2: %d\n",result2);
printf("Result3: %d\n",result3);
   return 0;
}

I want to identify that data hazard occurrence in the code from the stats file generated by gem5. Some references suggest to look at the following values from the gem5 stats file.
system.cpu.pipeline_stalls 
system.cpu.pipeline_stalls_mem
system.cpu.pipeline_stalls_reg

But when I open my stats file I could not find the values in the stats file. The general CPU model was selected
I am using gem5 24.1.0.0 and ubuntu 24.04
How can I get out of this problem?
Any sort suggestions will be helpful for me.
Thanks in advance.

**Comment by powerjg** [comment, marked as answer]
Maybe you want to use the O3CPU or the MinorCPU. This slide deck from the bootcamp may be helpful: https://bootcamp.gem5.org/#02-Using-gem5/04-cores. The video is available here: https://www.youtube.com/watch?v=vfj0-n6JQOE&list=PL_hVbFs_loVR_8ntTTmmG6YEq3Po_4snu&index=7
